articles/200_coexistence_and_migration.md

```diff
@@ -0,0 +1,247 @@
+  Additionally it covers the different aspects of migrating ROS packages from ROS 1 to ROS 2.
+author: '[Dirk Thomas](https://github.com/dirk-thomas)'
+published: true
+categories: Overview
+---
+
+{:toc}
+
+# {{ page.title }}
+
+<div class="abstract" markdown="1">
+{{ page.abstract }}
+</div>
+
+Original Author: {{ page.author }}
+
+## Goals
+
+### Separately usable
+
+* One version should not be affected by other versions choices / drawbacks.
+
+* ROS 1 users shouldn't be worried about ROS 2 development potentially negatively affecting ROS 1.
+
+  * No changes to existing ROS 1 packages should be necessary to continue to work.
+
+* ROS 2 users shouldn't need to make compromises due to ROS 1 choices / limitations / design decisions.
+
+### Support dual-homed applications
+
+To support creating dual-homed applications like the [ros1_bridge](https://github.com/ros2/ros1_bridge) ROS 1 and ROS 2 must be usable within a single application.
+
+### Minimize migration effort
+
+To ease the migration path the goal is to minimize the effort necessary where possible and when this goal doesn't conflict with the previously mentioned goals.
+
+## Technical Constraints
+
+### Namespace of ROS packages
+
+The first and very important question is about the namespace of ROS packages.
+Should ROS 1 and ROS 2 packages use separate namespaces or should they share the same namespace?
+For example can there be two packages names `foo` - one in ROS 1 and the other in ROS2 or do they have to be named differently.
+
+In the case that they use separate namespaces that does not prevent the release of a package `bar` into both, ROS 1 as well as ROS 2.
+
+Having them share a single namespace makes it difficult for users to distinguish them.
+
+Using separate namespaces allows e.g. to release a package `foo` in ROS 2 which provides the same API (but different implementation) as the ROS 1 package with the same name.
+This way downstream packages don't need to change when switching between the ROS 1 and ROS 2 version of the package.
+
+**Decision:** Use separate namespaces for packages in ROS 1 and ROS 2 (this is currently already the case).
+
+### Namespace of rosdep keys
+
+The next question is very similar and is about the namespace of the `rosdep` keys.
+Should ROS 1 and ROS 2 use a different set of `rosdep` rules or should they share the same database?
+
+#### Same key vs. different key
+
+If a ROS package declares a dependency on the `rosdep` key `python-foo` that maps on e.g. Ubuntu Xenial to the Python 2 package `foo`.
+In order to work for ROS 2 it needs to depend on the Python 3 version of the package instead.
+
+Currently the mapping in `rosdep` only depends on the platform and doesn't consider which ROS distribution is being used.
+
+If the migration to ROS 1 requires the key to be changed to `python3-foo` that implies that any package which depends on Python packages does need a modification in the manifest.
```

Comment:
"migration to ROS *2*"
And if you want to eliminate the possibility of ever moving ROS1 to Python3, that should be stated.

Reply:
This looks like a duplicate of https://github.com/ros2/design/pull/139#discussion_r135758429 ?